pydantic/dataclasses.py

```diff
@@ -178,7 +178,7 @@ def create_dataclass(cls: type[Any]) -> type[PydanticDataclass]:
 
         original_cls = cls
 
-        config_wrapper = _config.ConfigWrapper(config)
+        config_wrapper = _config.ConfigWrapper(config or getattr(cls, '__pydantic_config__', None) or None)
```

Comment:
```suggestion
        config_wrapper = _config.ConfigWrapper(getattr(cls, '__pydantic_config__', None) if config is None else config)
```
Otherwise the empty config would be ignored. I think if an empty dict is passed though that should signify something different than passing just `None`.

If that seems wrong for some reason feel free to reject this.

Reply:
Also, I'm not sure what the final `or None` is doing there, which is making we nervous that I'm misunderstanding something and there really is a good reason to convert anything falsy to `None`

Reply:
Yeah makes sense. I implemented it differently since this was going to trigger reformatting anyway. And added a test.